### DISCUSSION TITLE
Do we need to mesh non application name spaces too?

### DISCUSSION THREAD

**Discussion by praveenpattisapu** [linkerd/linkerd2#6033] [General] [answered]
Hi, we are using linkerd as the service mesh but we do have multiple name spaces in the cluster,is it a good practise to mesh all the applications in all the name spaces including gatekeeper-system, kube-node-lease, kube-public,kube-system and linkerd etc.

Any advise/suggestions here are highly appreciated.
All the applications name spaces are already meshed.

Thanks!!

**Comment by kleimkuhler** [comment, marked as answer]
@praveenpattisapu No you do not need to mesh non-application namespaces. The reasons that you install a service mesh in the first place—such as reliability, observability, or security—are not really things that have equal importance for non-application traffic in your cluster. You probably could mesh some of these namespaces without issue, but it's certainly not something that is required or considered a best-practice.
